delfin/drivers/manager.py

```diff
@@ -114,3 +120,65 @@ def _get_driver_cls(self, **kwargs):
         msg = "Storage driver '%s' could not be found." % name
         LOG.error(msg)
         raise exception.StorageDriverNotFound(name)
+
+    def _validate_drivers(self):
+        for name in self.names():
+            driver_cls = self[name].plugin
+            try:
+                # check if driver implements the StorageDriver interface
+                if not issubclass(driver_cls, StorageDriver):
+                    raise exception.BadStorageDriver(message="StorageDriver "
+                                                             "interface not "
+                                                             "implemented by "
+                                                             "%s driver"
+                                                             % driver_cls)
+                # check driver specification schema
+                driver_spec = self._validate_driver_specification(driver_cls)
+                # check for drivers feature set
+                self._validate_resource_metrics_support(driver_spec)
+            except Exception as ex:
+                self._handle_drivers_validation_error(driver_cls, ex)
+
+    def _handle_drivers_validation_error(self, driver_cls, ex):
+        LOG.error("Failed for %s driver with error %s" % (driver_cls, ex.msg))
+        # FIXME (Amit): Add Alert for driver's error
+        #   Also enable feature flag to make Northbound API aware
+        return
+
+    def _validate_driver_specification(self, driver_cls):
+        # get driver's resource metrics specifications
+        driver_spec = driver_cls.get_spec(context=context.
+                                          RequestContext(is_admin=True))
+        if driver_spec is None:
+            raise exception.BadStorageDriver("driver's capability feature "
+                                             "not supported")
+        try:
+            jsonschema.validate(driver_spec,
```

Comment:
this only validates the schema.. driver can give their own name for the metric.. 
for my modified  fake_driver I get reponse as below
![image](https://user-images.githubusercontent.com/45681499/109796297-fc240380-7c3d-11eb-941e-6315379056af.png)

Our idea is to validate drivers follow the same name format.. we should also give driver an option to reuse super template if they follow all metrics.. this may be challenging...

Reply:
I think concern about driver implementer using custom name is taken care by ```'additionalProperties': False``` check in defined template

Reply:
Yes, this check only gives an exception during the module loading.. 
```
Failed for <class 'delfin.drivers.hitachi.vsp.vsp_stor.HitachiVspDriver'> driver with error Implementation issues with Storage driver 'Driver's capability feature not supported': delfin.exception.BadStorageDriver: Implementation issues with Storage driver 'Driver's capability feature not supported'

```
I can still call /capability and get metrics from the driver which was not followed the spec.. 
I hope we should have this check while returning the spec also..